SRC/bank_000.asm

```diff
@@ -132,10 +132,20 @@ VBlankHandler: ;{ 00:0154
         call VBlank_updateMap
         jr .endIf_B
     .else_B:
+		;;;;m2maps: skip updating gameplay status bar if game is paused
+			ld a, [gameMode]
+			cp a, gameMode_paused
+			jr nz, .skipUpdateHudPaused
+				ld a, BANK(VBlank_updateStatusBarPaused)
+				ld [rMBC_BANK_REG], a
```

Comment:
moved to bank 10 - I moved the label as well since I didn't know if other stuff in the vanilla game needed the original label or jumped there. All of the status bar content has changed except the metroid counter. I could probably remove the vanilla label though if that's sensible, and put it back if that breaks things.

Reply:
I didn't mean that you're missing `VBlank_updateStatusBarPaused` the routine, I meant you're setting the bank as if you're going to do a `call VBlank_updateStatusBarPaused`, but there is no `call` :O

Reply:
> I didn't mean that you're missing `VBlank_updateStatusBarPaused` the routine, I meant you're setting the bank as if you're going to do a `call VBlank_updateStatusBarPaused`, but there is no `call` :O

I figured it out

The setup here preps this later call to run OAM_DMA:
```
    .else_B:
            ; m2maps: skip updating gameplay status bar if game is paused
                ld a, [gameMode]
                cp a, gameMode_paused
                jr nz, .skipUpdateHudPaused
                    ld a, BANK(m2maps_pauseAdjustSpriteSetup)
                    ld [rMBC_BANK_REG], a
                    jr .endIf_B
                .skipUpdateHudPaused:			
            ; end m2maps block
        ld a, BANK(VBlank_updateStatusBar)
        ld [rMBC_BANK_REG], a
        call VBlank_updateStatusBar
    .endIf_B:
		
; End vblank
    call OAM_DMA ; Sprite DMA
    ld a, [bankRegMirror]
    ld [rMBC_BANK_REG], a
    ld a, $01
    ldh [hVBlankDoneFlag], a
    pop hl
    pop de
    pop bc
    pop af
reti
```

I tried all configs of commenting this m2maps block out and basically if it's totally removed, you get the normal HUD overwriting the pauseHUD sprites that are needed to show in-game-time, current items out of total items, and the metroid remaining sprite.

That routine to display the map HUD is called in a different block that handles pausing. I think originally I tried to call it from here but I had to put the call elsewhere but the setup over here.

Reply:
I can understand jumping past the call to `VBlank_updateStatusBar` via the `jr .endIf_B`, but is the `ld a, BANK(m2maps_pauseAdjustSpriteSetup) : ld [rMBC_BANK_REG], a` actually doing anything?

Reply:
Was able to finally get this down to just the jump logic! And remove those placeholder NOPs we mentioned on discord.

Source of the bug was the title screen load I had modded to load a map at the title screen, thinking it was needed to have one available at start of gameplay at the landing site.

What actually happened was, lacking a sane currentLevelBank value, the game loaded gargabe data the size of a map to the window, overwrote the stack pointer as you described, and then returned to a location that accidentally worked.

Then, the door warp to the landing site reloaded the map (which I didn't realize was happening), making the initial map load of garbage data appear like it had functioned correctly.

After ripping out the initial map code in bank 5 I was able to remove the placeholder NOPs and have the game run again. I still have to step through it to verify now that I have a handle on the BGB debugger (which is actually useful as opposed to whatever VBA one I was using prior).

Thanks so much!!!

Reply:
Well diagnosed, any solution that removes code is a good solution :D